design/FY2021/NFP_AirflowNetwork_Specified_Flow_Element.md

```diff
@@ -0,0 +1,80 @@
+# AirflowNetwork Specified Flow Element #
+
+Jason DeGraw, Prateek Shrestha
+
+Oak Ridge National Laboratory
+
+Original Date: January 31, 2021
+
+## Justification for New Feature ##
+AirflowNetwork does not currently have a constant flow element linkage, which limits the degree to which
+certain elements (including supply/exhaust fans and infiltration) can be connected to AirflowNetwork.  
+
+## Overview ##
+AirflowNetwork has a number of ways to represent unintentional flows between zones, but no good way to
+represent flows of known magnitude. This limitation means that AirflowNetwork
+
+  * is not compatible with the various available interzonal flow objects,
+  * is not compatible with the various available infiltration objects, and
+  * cannot represent very simple fans very well.
+
+This new feature will begin to address these deficiencies. The Jacobian elements needed by
+AirflowNetwork’s solver is computed by the flow element “calculate” function:
+
+```
+NF = element.calculate(..., F, DF);
+```
+
+The constant flow with specified flow will return 1, and the Jacobian values are computed as follows:
+
+  * F is computed from the specified flow
+  * DF is zero
+
+## Approach ##
+This feature will be added through the execution of these subtasks:
+
+  * Create a new airflow element SpecifiedFlowRate
+  * Create a new input object AirflowNetwork:MultiZone:SpecifiedFlowRate
+  * Create necessary inputs to allow users to incorporate this element into models
+  * Create unit tests and documentation
+
+## Testing/Validation/Data Sources ##
+Unit test will be added to ensure numerical values are calculated properly.
+
+## Input Output Reference Documentation ##
+Additions will be made to the Input Output Reference for the SpecifiedFlowRate airflow element.
+
+## Input Description ##
+```
+AirflowNetwork:MultiZone:SpecifiedFlowRate,
+      \min-fields 2
+      \memo This object specifies the flow through a link.
+ A1 , \field Name
+      \required-field
+      \type alpha
+      \reference SurfaceAirflowLeakageNames
+      \note Enter a unique name for this object.
+ N1 , \field Air Flow Value
+      \type real
+      \required-field
+      \note Enter the air flow value for this element.
+ A2 ; \field Air Flow Units
+      \type choice
+      \key kg/s
+      \key scfm
+      \key slpm
+      \default kg/s
+      \note Enter the air flow units for this element.
```

Comment:
Units of airflow inputs in E+ are m3/s. Is it possible to be consistent with others?

Reply:
@lgu1234 Yes, we can do that. How about changing the choices to be kg/s and m3/s?

Reply:
For m3/s there's always the question of the density basis: outdoor current, indoor current, or standard? Another choice field? I've been considering such a choice field for the simple airflow objects (ZoneVentilation:* and ZoneInfiltration:*). Or maybe in this case it's just assumed to be standard density to align with HVAC system flow rate inputs?

Reply:
I guess `scfm` was going to be one of those. Anyways, not sure if the new editor will support it, but for field N1, you can add
```
      \unitsBasedOnField A2
```
Or just have two N fields:
```
A2, Air Flow Input Basis
  /key MassFlowRate
  /key VolumeFlowRateStandardDensity
  /key VolumeFlowRateOutdoorDensity
  /key VolumeFlowRateZoneDensity
N1, Air Mass Flow Rate
  /units kg/s
N2, Air Volume Flow Rate
  /units m3/s
```

Just thinking out loud.